content/blog/tikv-3.0ga-release-notes.md

```diff
@@ -0,0 +1,164 @@
+---
+title: TiKV 3.0 GA Release & Benchmarks
+date: 2019-07-08
+---
+
+Today we're proud to announce the general availability of TiKV 3.0! Before release, TiKV 3.0 underwent a rigorous testing regime, including the [official Jepsen test](https://www.pingcap.com/blog/tidb-passes-jepsen-test-for-snapshot-isolation-and-single-key-linearizability/) with TiDB.
+
+In this version, we tackled many problems of stability at massive scale. Whether spanning hundreds of nodes or storing over a trillion key-value pairs, we've seen our users put TiKV to the test in serious, real-world, production scenarios. In 3.0, we've applied the lessons learned from these deployments to bring a host of new features that can better support users' growing demands.
+
+## Steady at scale
+
+For the 3.0 release, we've improved TiKV by:
+
+* **Optimizing the Raft heartbeat mechanism.** With the [hibernate region](https://github.com/tikv/rfcs/blob/master/text/2019-03-04-hibernate-raft.md) feature, TiKV now adjusts the heartbeat frequency according to region activity. That means you'll see less CPU time and network traffic from idle regions.
+
+* **Distributing Garbage Collection.** The introduction of a distributed garbage collector improves performance on large scale clusters dramatically, leading to better stability through more consistent performance.
+
+* **Pessimistic Locking.** It's now possible to ask TiKV to enforce transactions using pessimistic locking. This means you can take exclusive ownership over a value for a duration, preventing other requests from modifying it.
+
+* **Expanding our coprocessor.** With lots of new or improved functionalities such as vector operations, batch executor, [RPN](https://en.wikipedia.org/wiki/Reverse_Polish_notation) functions, `work-stealing` thread pool model,  our coprocessor continues to evolve, allowing for more powerful and efficient ways to work with your data.
+
+* **Enhancing operator friendliness.** Human or machine, we've empowered our operators to get more out of TiKV by unifying our log format, adding new features to `tikv-ctl`, adding even more in-depth metrics, and serving HTTP based metrics. This makes TiKV easier to operate, inspect, and enjoy.
+
+* **Refining request types.** While TiKV previously supported commands like `BatchGet`, 3.0 brings a new `BatchCommands` request type. This allows TiKV to handle batches of requests of differing kinds (such as `Get` and `Put`), leading to less data on the wire and better performance. We also added support for raw reversed scanning, and `Insert` semantics on prewrite.
+
+* **Reducing write amplification.** Inspired by the great ideas from [WiscKey](https://www.usenix.org/system/files/conference/fast16/fast16-papers-lu.pdf), we implemented Titan, a key-value plugin that improves write performance for scenarios with value sizes greater than 1KB, and relieves write amplification in certain degrees.
```

Comment:
```suggestion
* **Reducing write amplification.** Inspired by [WiscKey](https://www.usenix.org/system/files/conference/fast16/fast16-papers-lu.pdf), this release includes [Titan], a key-value plugin that improves write performance for scenarios with value sizes greater than 1KB, and reduces write amplification.
```

Hyperlink Titan to a blog post.

Reply:
There is no titan blog post on tikv.org.

Reply:
I propose we port https://pingcap.com/blog/titan-storage-engine-design-and-implementation/ to tikv.org at a later date and publish it. @dcalvin and I are already working on this. We can add this hyperlink later.